src/services/CryptoFlowService.ts

```diff
@@ -54,11 +54,13 @@ export const executeAllowanceApprove = async(selectedQuote: ICryptoFlowQuote, ac
 
     if (!isTokenHbar) {
         const swapContract = JSON.parse(await getConfig("swapContract"));
-        const amount = Math.ceil(
-            Math.pow(10, sourceToken!.decimals!) * (
-                selectedQuote.source.amountExpected! * 1.2 // with a little buffer
+        const amount = approve
+            ? Math.ceil(
+                Math.pow(10, sourceToken!.decimals!) * (
+                    selectedQuote.source.amountExpected! * 1.2 // with a little buffer
+                )
             )
-        );
+            : 0;
```

Comment:
Here could be an issue if user already has an allowance for this token in place

Reply:
we decided to skip for now